Sources/NIOCore/Docs.docc/swift-concurrency.md

```diff
@@ -0,0 +1,244 @@
+# NIO and Swift Concurrency
+
+This article explains how to interface between NIO and Swift Concurrency 
+
+NIO was created before native Concurrency support in Swift existed, hence, NIO had to solve
+a few problems that have solutions in the language today. Since, the introduction of Swift Concurrency
+NIO has added numerous features to make the inter-op between NIOs channel eventing system and Swift's
+Concurrency primitives as easy as possible.
+
+### EventLoopFuture bridges
+
+The first bridges, that NIO introduced were adding methods on ``EventLoopFuture`` and ``EventLoopPromise``
+to bridge to and from Concurrency. Namely these bridges are ``EventLoopFuture/get()`` and ``EventLoopPromise/completeWithTask(_:)``.
+
+> Warning: The future ``EventLoopFuture/get()`` method does not support task cancellation.
+
+Here is a small example of how these work:
+
+```swift
+let eventLoop: EventLoop
+
+let promise = eventLoop.makePromise(of: Bool.self)
+
+promise.completeWithTask {
+    try await Task.sleep(for: .seconds(1))
+    return true
+}
+
+let result = try await promise.futureResult.get()
+```
+
+> Note: The `completeWithTask` method is creating an unstructured task under the hood.
```

Comment:
```suggestion
> Note: The `completeWithTask` method creates an unstructured task under the hood.
```

Reply:
Is it worth noting that the `Task` is returned to the caller as a `@discardableResult` so they can e.g. cancel if necessary?